### PR TITLE
Update Go version and CI deps

### PR DESCRIPTION
**Pull request by jamestelfer**
A number of changes to make CI work properly again and bring the Go version up to snuff.

- Go updated to 1.20
- `golangci-lint` now has a pinned version and an updated configuration to account for the changes to linters
- some code changes to account for linter changes
- `setup-go` version updated, and now uses `go.mod` to determine the required version of Go as well as the module cache
- `plugin-tester` image was upgraded with a new version of `bats-mock` that processes arguments differently. The version is now pinned and the tests have been upgraded.
- some GHA plugin versions were updated as the Node 12 versions are now deprecated.

### CODE REVIEW

File: .github/workflows/go-checks.yml [hunk 1]
```diff
@@ -28,37 +26,30 @@ jobs:
...
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
-      - name: Install Go
-        uses: actions/setup-go@v3
+      - uses: actions/setup-go@v3
         with:
-          go-version: ${{ env.GO_VERSION }}
+          go-version-file: src/go.mod
```
Comment (therealvio) [inline .github/workflows/go-checks.yml:33]: Big plus for this, as one using this for personal `golang` projects.

**Review by therealvio — APPROVED** [review]
Not managing deps using `renovate`
image

File: .github/workflows/go-checks.yml (hunk 1 above)
Comment (jamestelfer) [reply .github/workflows/go-checks.yml:33]: 💯

**Comment by jamestelfer** [comment]
Note that Renovate will come, we just needed an improvement sooner.

**Pull request merged**